source/unit/Constraints.ooc

```diff
@@ -31,6 +31,19 @@ Modifier: abstract class {
 	child: This = null
 	init: func
 	init: func ~parent (=parent)
+	free: override func {
+		if (this child) {
+			if (this child parent == this)
+				this child parent = null
+			this child free()
+		}
+		if (this parent) {
+			if (this parent child == this)
+				this parent child = null
```

Comment:
This and similar line above is to prevent circular `free` calls, and allow to clear entire hierarchy of objects from any point in the parent-child chain.